source/server/version.rc.in

```diff
@@ -0,0 +1,48 @@
+#if defined(__MINGW64__) || defined(__MINGW32__)
+    // MinGW-w64, MinGW
+    #if defined(__has_include) && __has_include(<winres.h>)
+        #include <winres.h>
+    #else
+        #include <afxres.h>
+        #include <winresrc.h>
+    #endif
+#else
+    // MSVC, Windows SDK
+    #include <winres.h>
+#endif
+
+LANGUAGE LANG_ENGLISH, SUBLANG_ENGLISH_US
+#pragma code_page(1252)
+
+VS_VERSION_INFO VERSIONINFO
+    FILEVERSION @GIT_TAG_MAJOR@,@GIT_TAG_MINOR@,@GIT_TAG_PATCH@,0
```

Comment:
If these variables are empty, are these version resources malformed? It seems safer to have CMakeLists.txt set these to 0 if they are empty.

Reply:
Updated the CMakeList.txt file to allow for optional patch number. Major and minor version is still mandatory. If the regex match fails, then we fail the generation and give error message that details required format of the Git tag.